client/modules/post/actions/postListActions.js

```diff
@@ -89,10 +116,94 @@ function receiveFeaturedPostList(json) {
 export function fetchFeaturedList() {
   return dispatch => {
     dispatch(requestFeaturedPostList())
-    return fetch('/api/posts')
+    return fetch('/api/posts/featured')
       .then(response => response.json())
-      .then(json =>
-        dispatch(receiveFeaturedPostList(json))
-      )
+      .then(json => {
+        if(json.success) {
+          var itemMap = {};
+          for(var i = 0; i < json.posts.length; i++) {
+            itemMap[json.posts[i]._id] = json.posts[i];
+          }
+          json.itemMap = itemMap;
+          return json;
+
+        } else {
+          //do something with the error
+          return json;
+        }
+      })
+      .then(json => dispatch(receiveFeaturedPostList(json)))
+  }
+}
+
+
+const shouldFetchList = (state, type) => {
+  console.log("------- CHECK SHOULD FETCH LIST -----------");
+  const postType = !type ? 'all' : type;
+  console.log(postType);
+  const items = state.post.list[postType];
+  console.log(items);
+  if(!items || items.length < 1) {
+    console.log("YES FETCH LIST");
+    return true
+  }
+  if(state.post.list.isFetching) {
+    console.log("FETCHING ALREADY DON'T FETCH LIST");
+    return false
+  }
+  return state.post.list.didInvalidate[postType]
+}
+
+
+export const fetchListIfNeeded = (postType) => (dispatch, getState) => {
```

Comment:
how do we pass variables in to this? a lot of our soflete stuff will be "fetchListBySquad(squadId)" for example